x-pack/plugins/apm/server/routes/fleet/get_latest_apm_package.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from 'kibana/server';
+import { APMPluginStartDependencies } from '../../types';
+import { APM_PACKAGE_NAME } from './get_cloud_apm_package_policy';
+
+export async function getLatestApmPackage({
+  fleetPluginStart,
+  request,
+}: {
+  fleetPluginStart: NonNullable<APMPluginStartDependencies['fleet']>;
+  request: KibanaRequest;
+}) {
+  const packageClient = fleetPluginStart.packageService.asScoped(request);
+  const { name, version } = await packageClient.fetchFindLatestPackage(
```

Comment:
I ended implementing something similar for synthtrace: https://github.com/elastic/kibana/pull/127257/files#diff-381f65aa3f9def6f1a385dff2de4bda5cb1924a86ae80963601026d4a2c789e8R56

There is a very handy `kibana=version` to filter only valid versions for the current running kibana version.  Does `fetchFindLatestPackage` take this into account?

Reply:
It looks like Fleet applies the kibana version as a query param in the implementation for `fetchFindLatestPackage`:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/epm/registry/index.ts#L86-L88=
https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/epm/registry/index.ts#L186=